crates/red_knot_module_resolver/src/db.rs

```diff
@@ -5,18 +5,24 @@ use crate::resolver::{
     internal::{ModuleNameIngredient, ModuleResolverSearchPaths},
     resolve_module_query,
 };
+use crate::supported_py_version::TargetPyVersion;
+use crate::typeshed::TypeshedVersions;
 
 #[salsa::jar(db=Db)]
 pub struct Jar(
     ModuleNameIngredient<'_>,
     ModuleResolverSearchPaths,
+    TargetPyVersion,
     resolve_module_query,
     file_to_module,
 );
 
-pub trait Db: salsa::DbWithJar<Jar> + ruff_db::Db + Upcast<dyn ruff_db::Db> {}
+pub trait Db: salsa::DbWithJar<Jar> + ruff_db::Db + Upcast<dyn ruff_db::Db> {
+    fn typeshed_versions(&self) -> &TypeshedVersions;
+}
```

Comment:
I think I would make a `typeshed_versions(db)` query instead. It should be possible to have zero-argument queries.